pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -234,13 +257,43 @@ public String segmentCommit(@QueryParam(SegmentCompletionProtocol.PARAM_INSTANCE
 
     final SegmentCompletionManager segmentCompletionManager = SegmentCompletionManager.getInstance();
     SegmentCompletionProtocol.Response response = segmentCompletionManager.segmentCommitStart(requestParams);
-    if (response.equals(SegmentCompletionProtocol.RESP_COMMIT_CONTINUE)) {
-      // Get the segment and put it in the right place.
-      boolean success = uploadSegment(multiPart, instanceId, segmentName, false) != null;
 
-      response = segmentCompletionManager.segmentCommitEnd(requestParams, success, false);
+    CommittingSegmentDescriptor committingSegmentDescriptor =
+        CommittingSegmentDescriptor.fromSegmentCompletionReqParams(requestParams);
+    boolean success = false;
+
+    if (response.equals(SegmentCompletionProtocol.RESP_COMMIT_CONTINUE)) {
+      File localTmpFile = null;
+      try {
+        // Get the segment from the form input and put it in the right place.
```

Comment:
Wrong comment. Get the segment and put it in a tmp area in the local file system